### ISSUE TITLE
False positive modifier_order violation when explicit order is not specified

### ISSUE THREAD

**Issue by biboran** [realm/SwiftLint#2435]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

False positive `modifier_order` violation is reported when `final` keyword precedes `public` keyword if order between two is not specified explicitly.

As far as I understand the rule it should only validate the ordering between keywords specified by the configuration keyword groups and leave unspecified declarations in place.

##### Complete output when running SwiftLint, including the stack trace and command used

Here is the config

```yml
whitelist_rules:
  - modifier_order

modifier_order:
  preferred_modifier_order:
    # - final #uncommenting this makes the violation go away
    - override
    - acl
```

Here is the output
```bash
echo 'final public class Foo { }' | swiftlint lint --no-cache --use-stdin
Loading configuration from '.swiftlint.yml'
<nopath>:1:14: warning: Modifier Order Violation: public modifier should be before final. (modifier_order)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
> 0.27.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
> building from source
* Which Xcode version are you using (check `xcode-select -p`)?
> 9.4.1

**Comment by jpsim** [comment]
Resolved in https://github.com/realm/SwiftLint/pull/2458
